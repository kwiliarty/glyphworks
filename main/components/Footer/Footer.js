import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import theme from '../../themes/glyphworks'

const StyledFooter = styled.footer`
  background-color: ${ props => props.backColor };
  color: ${ props => props.foreColor };
  padding: ${ props => props.theme.basePadding };
  font-size: ${ props => props.theme.smallFontSize };
`

const Footer = props => {

  const backColor = theme.colors[ props.backColor ] || props.backColor
  const foreColor = theme.colors[ props.foreColor ] || props.foreColor
  const date = new Date()

  return (
    <StyledFooter backColor={ backColor } foreColor={ foreColor }>
      Â©{ date.getFullYear() }
    </StyledFooter>
  )
}

Footer.propTypes = {
  /** The background color, the name of a color from the theme */
  backColor: PropTypes.string,
  /** The component's children */
  children: PropTypes.node,
  /** The foreground color, the name of a color from the theme */
  foreColor: PropTypes.string,
}

Footer.defaultProps = {
  backColor: 'lightBrown',
  foreColor: 'darkRed',
}

export default Footer
