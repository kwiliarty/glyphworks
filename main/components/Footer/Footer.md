The application Footer

```jsx
<Footer>
  Default Footer
</Footer>
```

```jsx
<Footer backColor='lightBrown' foreColor='darkRed'>
  Custom Footer
</Footer>
```
