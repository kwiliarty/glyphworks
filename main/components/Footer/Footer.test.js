import React from 'react'
import { render, screen } from '@testing-library/react'
import Footer from './Footer.js'

describe( 'The Header component', () => {

  beforeAll(() => {
    jest.useFakeTimers()
    jest.setSystemTime( new Date( '3023-04-01' ).getTime() )
  })

  afterAll(() => {
    jest.useRealTimers()
  })

  it( 'displays the copyright year', () => {
    render( <Footer /> )
    const footer = screen.getByRole( 'contentinfo' )
    expect( footer ).toHaveTextContent( '3023' )
    expect( footer ).toMatchSnapshot()
  })

  it( 'can accept custom colors', () => {
    render(
      <Footer foreColor='wheat' backColor='pink' />
    )
    const footer = screen.getByRole( 'contentinfo' )
    expect( footer ).toHaveStyleRule('color', 'wheat')
    expect( footer ).toHaveStyleRule('background-color', 'pink')
    expect( footer ).toMatchSnapshot()
  })
})
