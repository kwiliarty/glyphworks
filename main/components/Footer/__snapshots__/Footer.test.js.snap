// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component can accept custom colors 1`] = `
.c0 {
  background-color: pink;
  color: wheat;
}

<footer
  class="c0"
>
  ©
  3023
</footer>
`;

exports[`The Header component displays the copyright year 1`] = `
.c0 {
  background-color: #c9b978;
  color: #621818;
}

<footer
  class="c0"
>
  ©
  3023
</footer>
`;
